### ISSUE TITLE
🐛 BUG: Importing conditionally script causes an error

### ISSUE THREAD

**Issue by OskarGrzelak** [withastro/astro#3180]
### What version of `astro` are you using?

1.0.0-beta.14

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

There is a problem with conditional importing script.

For example:

```
---
  const {loadScript} = Astro.props;
---
/* some html */
<script>
  import "./alwaysLoadedScript.js"
</script>
{ loadScript &&
  <script>
    import "./sometimesLoadedScript.js"
  </script> 
}
```

![Bez tytułu](https://user-images.githubusercontent.com/36723922/164706812-e8b64805-9a01-41ac-959d-15298fe3db69.jpg)

or

```
---
const { loadScript } = Astro.props;
---

<script src="./alwaysLoadedScript.js" />

{ loadScript && <script src="./sometimesLoadedScript.js" /> }
```

![err](https://user-images.githubusercontent.com/36723922/164706821-a2b7902a-b035-456a-b097-00ab2eddb4b4.jpg)

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-chkxnu?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by cshaa** [comment]
That is nice, but if `is:raw` is used, the import path is not transformed on build, and therefore this is **not** an actual fix, and will only work in dev mode.

**Comment by matthewp** [comment]
@cshaa Correct, there is no way to conditionally a script that's in your src folder in production.
